source/MaterialXView/Material.cpp

```diff
@@ -301,13 +301,23 @@ void Material::bindViewInformation(const mx::Matrix44& world, const mx::Matrix44
     }
 }
 
+void Material::unbindImages(mx::GLTextureHandlerPtr imageHandler)
+{
+    for (auto filePath : _boundImages)
+    {
+        imageHandler->unbindImage(filePath);
+    }
+}
+
 void Material::bindImages(mx::GLTextureHandlerPtr imageHandler, const mx::FileSearchPath& searchPath, const std::string& udim)
 {
     if (!_glShader)
     {
         return;
     }
 
+    _boundImages.clear();
```

Comment:
Viewer materials now keep track of what images are bound here and clears them after draw. We we're basically filling up the slots for for images whether used or not across all materials and ending up running out of slots very quickly.